lesson1/src/arrays/arrayProblems.js

```diff
@@ -1,46 +1,51 @@
 class ArrayProblems {
 
-  /*
-   This method should return true if a given Array (array)
-   has all equal elements and false otherwise
-   Eg1: [1,1,1] -> true
-   Eg2: [1,1,1,2] -> false
-   */
   static allEqualElements(array) {
     for (let i = 0; i < array.length; i++) {
       for (let j = i; j < array.length; j++) {
-        // todo: do stuff here
+        if (array[i] != array[j]) {
+          return false;
+        }
       }
     }
     return true;
   }
 
-  /*
-   This method should return the maximum element from an integer Array (array)
-   Eg1: [1,1,1] -> 1
-   Eg2: [-3,-1,0,5] -> 5
-   */
   static getMax(array) {
-    // todo: do stuff here
+    if (array) {
+      array.sort(function (n1, n2) {
+        return n1 - n2;
+      });
+      return array.pop();
+    } else {
+      return -1;
+    }
   }
 
-  /*
-   Can you implement getMax again without using any js collection method?
-   */
   static getMax2(array) {
-    // todo: 🙌 do magic !
+    let temp = array[0];
+    let i = 0;
+    if (array.length != 0) {
+      for (i; i < array.length; i++) {
+        if (temp >= array[i]) {
+          console.log(array[i]);
+          temp = array[i];
+        }
+      }
+    }
+    if (temp < 0) {
+      temp = -1;
+    }
+    return temp;
   }
 
-  /*
-   A left rotation operation on an array shifts each
-   of the array's elements 1 unit to the left.
-   For example, if 2 left rotations are performed on array [1,2,3,4,5],
-   then the array would become [3,4,5,1,2].
-
-   This method should perform n left rotations on the array
-   */
-  static rotateLeft(array, n){
-    // todo: 🙌 do magic !
+  static rotateLeft(array, n) {
+    for (let i = 0; i < n; i++) {
+      let temp = array[0];
```

Comment:
Always take a second look for how your algorithm behaves on edge cases, what happens when we want to rotate an empty array?